deps/wazuh_testing/wazuh_testing/tools/sources/report_generator.py

```diff
@@ -0,0 +1,366 @@
+                    │       │   ├── binaries
+                    │       │   └── stats
+                    │       └── logs
+                    └── workers
+                        └── worker1
+                            ├── data
+                            │   ├── binaries
+                            │   └── stats
+                            └── logs
+                """)
+                raise ValueError
+
+        self.component_path = {
+            'agents': agents_path,
+            'managers': managers_path,
+            'master': master_path,
+            'workers': worker_path
+        }
+
+        if os.path.isdir(worker_path):
+            self.n_workers = len(os.listdir(worker_path))
+            self.cluster_environment = True
+        else:
+            self.n_workers = 0
+            self.cluster_environment = False
+
+        self.n_agents = len(os.listdir(agents_path))
+
+    def get_instances_artifacts(self, component, hosts_regex=".*"):
+        artifacts_files = []
+        if component == 'all':
+            artifacts_files += self.get_instances_artifacts('managers', hosts_regex)
+            artifacts_files += self.get_instances_artifacts('agents', hosts_regex)
+        elif component == 'managers':
+            artifacts_files += self.get_instances_artifacts('master', hosts_regex)
+            if self.cluster_environment:
+                artifacts_files += self.get_instances_artifacts('workers', hosts_regex)
+        else:
+            artifact_path = self.component_path[component]
+            artifacts_files = [{"name": f, "path": os.path.join(artifact_path, f)}
+                               for f in os.listdir(artifact_path) if re.match(rf'{hosts_regex}', f)]
+        return artifacts_files
+
+    def get_instance_all_log_files(self, hostname):
+        files = os.path.join(self.get_instances_artifacts('all', hostname)[0]['path'], 'logs')
+        host_log_list = [log for log in os.listdir(files)]
+        return host_log_list
+
+    def get_instances_logs(self, log, component, hosts_regex=".*"):
+        artifacts_paths = self.get_instances_artifacts(component, hosts_regex)
+        for host_artifact_path in artifacts_paths:
+            host_artifact_path['logs'] = {}
+            log_files = self.get_instance_all_log_files(host_artifact_path['name']) if log == 'all' else [log]
+
+            for log_file in log_files:
+                general_path = os.path.join(host_artifact_path['path'], 'logs')
+                host_artifact_path['logs'][log_file] = os.path.join(general_path, log_file)
+            del host_artifact_path['path']
+
+        return artifacts_paths
+
+    def get_instances_process_metrics(self, process, component, hosts_regex='.*'):
+        files = self.get_instances_artifacts(component, hosts_regex=hosts_regex)
+        for file in files:
+            file.update((k, os.path.join(v, 'data', 'binaries', process + '.csv')) for k, v in file.items())
+        return files
```

Comment:
We should use more descriptive names, instead of `k` and `v`

Reply:
Done [7448bb7](https://github.com/wazuh/wazuh-qa/pull/2502/commits/7448bb7edc3e9506d01084be4e384b48ceda7a48)